From 9e0f5b11a5a329fecd972afaf7dfaf40490d6497 Mon Sep 17 00:00:00 2001
From: Nikolaus Rath <Nikolaus@rath.org>
Date: Thu, 8 Oct 2015 12:24:36 -0700
Subject: Restore backwards compatibility of setattr() handler

Forwarded: yes
Last-Update: 2015-09-10

The 0.41.1 accidentally broke backwards compatibility by introducing
default values other than None for the EntryAttributes class.
Some file systems were relying on every attribute of the EntryAttributes
instance passed to setattr() to be None if the attribute was not
supposed to be updated (as opposed to only looking at those attributes
that are actual candidates for an update).

This patch restores backwards compatibility.

Patch-Name: setattr_compat_patch.diff
---
 src/llfuse/handlers.pxi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/llfuse/handlers.pxi b/src/llfuse/handlers.pxi
index 09b006f..5b94392 100644
--- a/src/llfuse/handlers.pxi
+++ b/src/llfuse/handlers.pxi
@@ -89,6 +89,11 @@ cdef void fuse_setattr (fuse_req_t req, fuse_ino_t ino, c_stat *stat,
 
     try:
         attr = EntryAttributes()
+
+        # For backwards compatibility
+        for name in attr.__slots__:
+            setattr(attr, name, None)
+
         if to_set & (FUSE_SET_ATTR_ATIME_NOW | FUSE_SET_ATTR_MTIME_NOW):
             ret = clock_gettime(CLOCK_REALTIME, &now)
             if ret != 0:
